### PR TITLE
CI: Merge doctest to example test in FlexCI

### PR DESCRIPTION
**Pull request by kmaehashi**
Let's test example snippets in documents within example tests. This is to avoid the overhead of launching additional FlexCI instances.

### CODE REVIEW

File: .pfnci/matrix.yaml [hunk 1]
```diff
@@ -230,32 +230,6 @@
...
-  system: "linux"
-  os: "ubuntu:20.04"
-  cuda: "11.4"
-  rocm: null
-  nccl: "2.10"
-  cutensor: "1.3"
-  cusparselt: "0.1.0"
-  cudnn: "8.2"
-  python: "3.9"
-  numpy: "1.21"
```
Comment (toslunar) [inline .pfnci/matrix.yaml:252]: python 3.9 -> 3.7, numpy 1.21 -> 1.20 after the change.  Could you confirm these are what you want?

Comment (kmaehashi) [reply .pfnci/matrix.yaml:252]: I think running examples/doctests with minimum supported python version is actually better 😃  Added comment.

**Review by toslunar — APPROVED** [review]

**Comment by kmaehashi** [comment]
kicked test: https://ci.preferred.jp/cupy.linux.cuda-example/84472/

**Comment by toslunar** [comment]
It seems `cupy.array_api` was not found in `refs/pull/6036/head`.  I'll try to kick the CI with `refs/pull/6036/merge`

```
00:18:13.321350 STDOUT 2430] 	+ SPHINXOPTS=-W 	
00:18:13.321361 STDOUT 2430] 	+ make doctest 	
00:18:13.324796 STDOUT 2430] 	sphinx-build -b doctest -d build/doctrees  -W source build/doctest 	
00:18:13.622317 STDOUT 2430] 	Running Sphinx v4.1.2 	1.4s
00:18:15.026895 STDOUT 2430] 	making output directory... done 	
00:18:15.036132 STDOUT 2430] 	[autosummary] generating autosummary for: contribution.rst, index.rst, install.rst, license.rst, overview.rst, reference/_deprecated.rst, reference/array_api.rst, reference/array_api_array.rst, reference/array_api_functions.rst, reference/binary.rst, ..., user_guide/basic.rst, user_guide/compatibility.rst, user_guide/cuda_api.rst, user_guide/difference.rst, user_guide/fft.rst, user_guide/index.rst, user_guide/interoperability.rst, user_guide/kernel.rst, user_guide/memory.rst, user_guide/performance.rst 	
00:18:15.162694 STDOUT 2430] 	Failed to import 'cupy.array_api': no module named cupy.array_api 	
00:18:15.162701 STDOUT 2430] 		
00:18:15.162708 STDOUT 2430] 	Warning, treated as error: 	
00:18:15.162889 STDOUT 2430] 	[autosummary] failed to import 'cupy.array_api._array_object.Array': no module named cupy.array_api._array_object.Array 	
00:18:15.383256 STDOUT 2430] 	make: *** [Makefile:176: doctest] Error 2
```

**Comment by kmaehashi** [comment]
Ah, I forgot that array api requires Python 3.8... will update requirements.

**Comment by kmaehashi** [comment]
Rebased, and fixed to use Python 3.8.

**Review by toslunar — APPROVED** [review]

**Comment by kmaehashi** [comment]
Merging as approved.

**Pull request merged**